client/extensions/woocommerce/app/promotions/promotion-models.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * External dependencies
+ */
+import { translate as moduleTranslate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import CurrencyField from './fields/currency-field';
+import PercentField from './fields/percent-field';
+import TextField from './fields/text-field';
+
+export function createModels( translate ) {
+	// -------------------------
+	// Reusable Promotion Fields
+	// -------------------------
+	const couponCode = {
```

Comment:
It wasn't clear until a few read-throughs that these are fields, not models themselves. Maybe add `Field` to the variable name?

Reply:
These are designed to be 1:1 with the promotions object, but I suppose I could add an extra property to them to denote the mapping...

Reply:
You mean you want the name `couponCode` to match the name in the model? It would be much clearer to use `couponCode: couponField` or something like that, especially with the ones where you're extending the object (like `fixedDiscount`)

Reply:
I ended up renaming the variables here and just mapped them in the objects below to be clearer. I also added JSDocs.